# HG changeset patch
# User Andreas Weber <octave@josoansi.de>
# Date 1546450472 -3600
#      Wed Jan 02 18:34:32 2019 +0100
# Branch stable
# Node ID 45d5a4ae636b1f3f6c3ede460f38147457af1cd6
# Parent  2398d2cc3d704132cb41f58333918325d09d3440
ov-java.cc: Bugfix use xstring_value for extraction

This is basically the same problem as detected V601 on bug #55347
but not part of that list.

diff --git a/libinterp/octave-value/ov-java.cc b/libinterp/octave-value/ov-java.cc
--- a/libinterp/octave-value/ov-java.cc
+++ b/libinterp/octave-value/ov-java.cc
@@ -3157,17 +3157,17 @@ equivalent
 @seealso{__java_set__, javaMethod, javaObject}
 @end deftypefn */)
 {
 #if defined (HAVE_JAVA)
 
   if (args.length () != 2)
     print_usage ();
 
-  std::string name = args(1).string_value ("__java_get__: NAME must be a string");
+  std::string name = args(1).xstring_value ("__java_get__: NAME must be a string");
 
   initialize_java ();
 
   JNIEnv *current_env = thread_jni_env ();
 
   octave_value retval;
 
   if (args(0).isjava ())
